src/Microsoft.Azure.WebJobs.Host/Executors/JobHostConfigurationExtensions.cs

```diff
@@ -276,6 +276,7 @@ public static ServiceProviderWrapper CreateStaticServices(this JobHostConfigurat
                 }
 
                 IFunctionIndex functions = await functionIndexProvider.GetAsync(combinedCancellationToken);
+                ApplyConfigurationToFunctionIndex(functions);
```

Comment:
All the current disable code we have is in FunctionIndexer.FromMethod. Why can't your new logic just slot in there? E.g. we already apply DisableAttribute and disabled settings coming from that attribute to the FunctionDescriptor there. Seems like your new additional setting check can be applied there?